### ISSUE TITLE
Non-vnode children not supported with shallow rendering

### ISSUE THREAD

**Issue by wnayes** [preactjs/enzyme-adapter-preact-pure#106]
If I try to shallow render a component that makes use of `children` in an unusual way (a "render prop" pattern for example) I encounter errors. Specifically, I hit Preact's `"Objects are not valid as a child. Encountered an object with the keys..."` message.

An example scenario is something like this:

```
function MyComponent(props) {
    return (
        <RenderPropExample>
            {() => <div>Example</div>}
        </RenderPropExample>
    );
}

function RenderPropExample(props) {
    return props.children();
}

const wrapper = shallow(<MyComponent />);
```

The issue appears to be that the [ShallowRenderStub](https://github.com/preactjs/enzyme-adapter-preact-pure/blob/master/src/shallow-render-utils.ts#L67) unconditionally returns its children.

In the example above, the ShallowRenderStub created for `RenderPropExample` would end up returning the render prop child function, which is not what the real `RenderPropExample` would do.

Preact then sees the non-vnode children that were returned by the ShallowRenderStub, and throws its debug error.

If I patch the adapter locally to make the ShallowRenderStub be a bit more picky about whether to return `children`, the issue goes away. (I need to then `dive()` through to get the render prop to execute, but that seems reasonable.) I'm not sure exactly what the perfect check would be, but I think even an approximate check would go a long way to avoid this in most practical cases.

**Comment by robertknight** [comment]
Preact's definition of an invalid vnode is any truthy value with an [undefined `type` field](https://github.com/preactjs/preact/blob/469da09b027edf2ed398c2210316f2cf816215b1/debug/src/debug.js#L280). I think it would be reasonable to use the same check in `ShallowRenderStub` and return `null` in that case.

**Comment by robertknight** [comment]
I've created a PR that should resolve this: https://github.com/preactjs/enzyme-adapter-preact-pure/pull/107

Would you be able to check out that branch and confirm whether it works for your particular scenario?

**Comment by wnayes** [comment]
The `isRenderable` logic there addresses the render prop case I mentioned here.

The [Preact guide says](https://preactjs.com/guide/v10/components/#fragments) that components can return arrays, which I think might not pass the check here, unless there is some normalization occurring somewhere else. Personally that isn't a problem for me, but it may be something to consider. My understanding is that it is probably always safer to err on the side of not rendering questionable children; the only drawback is extra `dive` calls to get more children to render.

**Comment by robertknight** [comment]
Hello,

This has been fixed in v2.2.1 by https://github.com/preactjs/enzyme-adapter-preact-pure/pull/107. Please give it a try and let me know if you still encounter issues.
